### PR TITLE
 fix the error logs have been omitted because of retryCount never exc…

### PR DESCRIPTION
**Pull request by DanielQujun**
**Description**
when logger.Error in `connectLoadInitFormats` function, there're retryCount >= 10 condtion
https://github.com/minio/minio/blob/2ee337ead5784d60971973b5399c20ebd9fdce44/cmd/prepare-storage.go#L188
```
	for i, err := range errs {
		if err != nil {
			if err == errDiskNotFound && retryCount >= 10 {
				logger.Error("Unable to connect to %s: %v", endpoints[i], isServerResolvable(endpoints[i], time.Second))
			} else {
				logger.Error("Unable to use the drive %s: %v", endpoints[i], err)
			}
		}
	}
```
which is reseted in https://github.com/minio/minio/blob/2ee337ead5784d60971973b5399c20ebd9fdce44/cmd/prepare-storage.go#L324
```
	for {
		select {
		case <-ticker.C:
			if tries == 10 {
				// Reset the tries count such that we log only for every 10 retries.
				tries = 1
			}

```
---
**Motivation and Context**

I found this when figuring out this issue: https://github.com/minio/minio/issues/14267
according to the context, I guess that we might want to print these errors at the last retry, so I change it to 9.

### CODE REVIEW

File: cmd/prepare-storage.go [hunk 1]
```diff
@@ -185,7 +185,7 @@ func connectLoadInitFormats(retryCount int, firstDisk bool, endpoints Endpoints,
 
 	for i, err := range errs {
 		if err != nil {
-			if err == errDiskNotFound && retryCount >= 10 {
+			if err == errDiskNotFound && retryCount >= 9 {
```
Comment (vadmeste) [inline cmd/prepare-storage.go]: Can you replace retryCount parameter with verboseLogging as boolean ?

**Review by vadmeste — COMMENTED** [review]
One comment

Thanks @DanielQujun

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/prepare-storage.go (hunk 1 above)
Comment (DanielQujun) [reply cmd/prepare-storage.go]: Hi, vadmeste, thx for your comment. I added a new commit, not sure is the right way, would pls take a look?

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — CHANGES_REQUESTED** [review]
The logic is wrong here @DanielQujun - we only wish to log once every 10 iterations not sure what is not correct in the current logic.

**Comment by DanielQujun** [comment]
> The logic is wrong here @DanielQujun - we only wish to log once every 10 iterations not sure what is not correct in the current logic.

@harshavardhana , Hi, I changed a bit of the code for making the logic clear, PTL.

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/prepare-storage.go [hunk 2]
```diff
@@ -321,12 +322,16 @@ func waitForFormatErasure(firstDisk bool, endpoints Endpoints, poolCount, setCou
...
 		case <-ticker.C:
-			if tries == 10 {
+			if tries >= 10 {
+				// Only log once every 10 iterations
+				verboseLogging = true
 				// Reset the tries count such that we log only for every 10 retries.
 				tries = 1
+			} else {
+				verboseLogging = false
 			}
```
Comment (harshavardhana) [inline cmd/prepare-storage.go]: Please rewrite it like this

```
verboseLogging = tries >= 10
if verboseLogging {
   tries = 1
}
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]
This would ensure a more clearer way to read the code.

File: cmd/prepare-storage.go (hunk 2 above)
Comment (DanielQujun) [reply cmd/prepare-storage.go]: wow, it looks elegant! thanks~

Comment (DanielQujun) [reply cmd/prepare-storage.go]: updated, and have squashed the commits

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Review by klauspost — APPROVED** [review]

**Pull request merged**